classes/Api/Firebase/Token.php

```diff
@@ -41,14 +41,37 @@ public function refresh()
         $response = $this->post([
             'json' => [
                 'grant_type' => 'refresh_token',
-                'refresh_token' => \Configuration::get('PS_PSX_FIREBASE_REFRESH_TOKEN'),
+                'refresh_token' => \Configuration::get(
+                    'PS_PSX_FIREBASE_REFRESH_TOKEN',
+                    null,
+                    null,
+                    (int) \Context::getContext()->shop->id
+                ),
             ],
         ]);
 
         if (true === $response['status']) {
-            \Configuration::updateValue('PS_PSX_FIREBASE_ID_TOKEN', $response['body']['id_token']);
-            \Configuration::updateValue('PS_PSX_FIREBASE_REFRESH_TOKEN', $response['body']['refresh_token']);
-            \Configuration::updateValue('PS_PSX_FIREBASE_REFRESH_DATE', date('Y-m-d H:i:s'));
+            \Configuration::updateValue(
+                'PS_PSX_FIREBASE_ID_TOKEN',
+                $response['body']['id_token'],
+                false,
+                null,
+                (int) \Context::getContext()->shop->id
+            );
+            \Configuration::updateValue(
+                'PS_PSX_FIREBASE_REFRESH_TOKEN',
+                $response['body']['refresh_token'],
+                false,
+                null,
+                (int) \Context::getContext()->shop->id
+            );
+            \Configuration::updateValue(
+                'PS_PSX_FIREBASE_REFRESH_DATE',
+                date('Y-m-d H:i:s'),
+                false,
+                null,
+                (int) \Context::getContext()->shop->id
```

Comment:
What do you think about having an adapter for `Configuration` in `classes/Adapter`, allowing you to reduce the number of parameters in all other places of the project?

Reply:
We don't have container yet to reuse instance for this type of Service.
Each time we need to use it, we have to instantiate it and if we use namespace, we have to add complete namespace in all place we use it in main module class and upgrade files...
Currently others adapters are instantiated more than one time at each execution and i'm not satisfied of that too.
I know it's boring to give each parameters to Configuration method but I don't find more faster to call an Adapter to do that with full namespace in main class.
I think we have to handle this kind of needs we a huge reflexion about architecture we need to use.

Reply:
I totally agree, having an equivalent of a container providing dependencies to each class of this project would be helpful.
I suggest to work on this idea because even if you did everything well in this PR, we can't be sure that the next person modyfing this project will think about using all the parameters.

Reply:
This feature is in top priority, so I suggest to handle architecture work on another pull request.
It's a very interesting topic and need huge work, maybe to be coupled with work on exceptions

Reply:
Indeed this would require too many changes and work, this has to be done in another PR. :)